backend/public-graph/graph/resolver.go

```diff
@@ -2695,15 +2720,21 @@ func (r *Resolver) ProcessPayload(ctx context.Context, sessionSecureID string, e
 			log.WithContext(osCtx).Error(e.Wrap(err, "error updating session in opensearch"))
 			return err
 		}
+		if err := r.IndexSessionClickhouse(osCtx, sessionObj); err != nil {
+			return err
+		}
 	}
 
-	if sessionHasErrors {
+	if sessionHasErrors && (sessionObj.HasErrors == nil || !*sessionObj.HasErrors) {
 		if err := r.OpenSearch.UpdateAsync(osCtx, opensearch.IndexSessions, sessionObj.ID, map[string]interface{}{
 			"has_errors": true,
 		}); err != nil {
 			log.WithContext(osCtx).Error(e.Wrap(err, "error setting has_errors on session in opensearch"))
 			return err
 		}
+		if err := r.IndexSessionClickhouse(osCtx, sessionObj); err != nil {
```

Comment:
mb thought I commented here already - this isn't a concern because we're using the `ReplacingMergeTree` in Clickhouse which will deduplicate by only keeping the most recent session row